src/Controls/src/Core/Button/Button.Android.cs

```diff
@@ -1,13 +1,41 @@
 ﻿#nullable disable
-using System;
-using System.Collections.Generic;
-using System.Text;
+using Google.Android.Material.Button;
 using Microsoft.Maui.Controls.Platform;
 
 namespace Microsoft.Maui.Controls
 {
 	public partial class Button
 	{
+		MaterialButton _materialButton;
+
+		private protected override void OnHandlerChangedCore()
+		{
+			base.OnHandlerChangedCore();
```

Comment:
This isn't going to always fire. Can we make this work without needing to subscribe to LayoutChange from the controls code?

Reply:
We are also unsubscribing when the parent Window is null. I have added a test to validate leaks. What would be a possible scenario for testing?